index.html

```diff
@@ -3879,13 +3882,10 @@ <h3>The <dfn>next()</dfn> method</h3>
             Return a {{Promise}} |promise:Promise| and execute the next steps [=in parallel=].
           </li>
           <li>
-            If the <a href="#dom-thingdiscovery-active">active</a> property is `true`, wait until the <a>discovery results</a> <a>internal slot</a> is not empty.
-          </li>
-          <li>
-            If <a>discovery results</a> is empty and the <a href="#dom-thingdiscovery-active">active</a> property is `false`, set the <a href="#dom-thingdiscovery-done">done</a> property to `true` and reject |promise|.
+            If the <a href="#dom-thingdiscovery-active">active</a> property is `true`, call the `next` method on the Discovery object's Symbol.AsyncIterator.
           </li>
           <li>
-            Remove the first {{ThingDescription}} object |td| from <a>discovery results</a>.
+            If the `value` property in the yielded result from the Symbol.AsyncIterator is undefined and the `done` property is set to `true`, set the <a href="#dom-thingdiscovery-done">done</a> property to `true` and reject |promise|.
```

Comment:
We need at least one example for how to handle async iterators in client code.

Reply:
I adjusted the first discovery example in 9646fb627c83b257cc671e70d4938ed0767d932d, I hope this points into the right direction.

Reply:
Fair.